crates/compiler/parse/src/expr.rs

```diff
@@ -2702,42 +2702,132 @@ pub fn record_field<'a>() -> impl Parser<'a, RecordField<'a>, ERecord<'a>> {
         and!(
             specialize_err(|_, pos| ERecord::Field(pos), loc!(lowercase_ident())),
             and!(
-                spaces(),
-                optional(either!(
-                    and!(byte(b':', ERecord::Colon), record_field_expr()),
-                    and!(
-                        byte(b'?', ERecord::QuestionMark),
-                        spaces_before(specialize_err_ref(ERecord::Expr, loc_expr(false)))
-                    )
+                optional(and!(
+                    and!(backtrackable(spaces()), byte(b':', ERecord::Colon)),
```

Comment:
Hmm, I'm surprised you need to do `backtrackable(spaces())`. The only errors that `spaces` will give is for things like having a tab character, or a carriage return without a newline, etc. And `backtrackable` doesn't do anything if the inner parser returns Ok. Do you have an example of where the `backtrackable` helps?

Reply:
Yeah, when I take out the `backtrackable`, the `default` test fails The crux of it comes down to the `optional`. 

```rust
optional(and!(
    and!(spaces(), byte(b':', ERecord::Colon)),
    record_field_expr()
))
```

When parsing `x ? 0.0`, `spaces()` returns `Ok(MadeProgress)`, but `byte` returns `Err(NoProgress)`, so `and!` returns `Err(MadeProgress)`. This ends up getting passed back until `record_field` returns that same `Err`. When `backtrackable` is used `backtrackable(spaces())` always returns `Ok(NoProgress)`. When combined in the `and!`, the result is now `Err(NoProgress)`, which `optional` converts to `Ok(None)`.